### PR TITLE
Merge mat stack fix into 2.1

### PR DESCRIPTION
**Pull request by Reonu**
Come on guys, this is the most important optimisation BY FAR and it's being held back because of some unrotated envmaps? That's something that should hold back the 2.1 release, but not the merge into 2.1.

Let's merge this now

### CODE REVIEW

**Comment by arthurtilly** [comment]
reonu when he sus

File: Makefile [hunk 1]
```diff
@@ -423,8 +426,8 @@ DEP_FILES := $(O_FILES:.o=.d) $(LIBZ_O_FILES:.o=.d) $(GODDARD_O_FILES:.o=.d) $(B
 # detect prefix for MIPS toolchain
 ifneq ($(call find-command,mips64-elf-ld),)
   CROSS := mips64-elf-
-else ifneq ($(call find-command,mips-n64-ld),)
-  CROSS := mips-n64-
+# else ifneq ($(call find-command,mips-n64-ld),)
+#   CROSS := mips-n64-
```
Comment (thecozies) [inline Makefile:430]: can this be removed?

File: src/game/debug_box.c [hunk 2]
```diff
@@ -102,8 +102,8 @@ Gfx dl_debug_cylinder_verts[] = {
 	gsSPEndDisplayList(),
 };
 
-u8 hitboxView  = FALSE;
-u8 surfaceView = FALSE;
+u8 hitboxView  = TRUE;
+u8 surfaceView = TRUE;
```
Comment (thecozies) [inline src/game/debug_box.c:106]: some debug leftovers i reckon

File: src/game/rendering_graph_node.c [hunk 3]
```diff
@@ -1245,6 +1306,10 @@ void geo_process_root(struct GraphNodeRoot *node, Vp *b, Vp *c, s32 clearColor)
 
         gDisplayListHeap = alloc_only_pool_init(main_pool_available() - sizeof(struct AllocOnlyPool), MEMORY_POOL_LEFT);
         initialMatrix = alloc_display_list(sizeof(*initialMatrix));
+        gCurLookAt = (LookAt*)alloc_display_list(sizeof(LookAt));
+        bzero(gCurLookAt, sizeof(LookAt));
+        gCurLookAt->l[1].l.col[1] = 0x80;
+        gCurLookAt->l[1].l.colc[1] = 0x80;
```
Comment (thecozies) [inline src/game/rendering_graph_node.c:1312]: just curious, why is this color needed?

**Review by thecozies — COMMENTED** [review]
a couple tiny things and a question

**Review by thecozies — APPROVED** [review]
once the little things are addressed it looks good to me

**Pull request merged**